### PR TITLE
Add actions for more services - addons and database

### PR DESCRIPTION
**Pull request by hemant-hc**
This PR introduces:

2 more services namely addons and database along with their tests and associated types.

Other changes:
1. fix some flaky tests
2. misc enhancements - use unique name for created entities (some actions) to avoid collision in DB in local testing, strip extra slashes from path

### CODE REVIEW

**Comment by hemant-hc** [comment]
@sneridagh some of the tests seem to be encountering random timeout errors, while they are working perfectly fine on my machine. I am working on debugging the issue. Any thoughts as to why this could happen?

**Comment by sneridagh** [comment]
Maybe a gh hickup... I restarted the job manually

**Comment by hemant-hc** [comment]
@sneridagh I have removed `groups`, `users`, and `history` actions from this PR because of the timeout issue. Tried a couple of things to debug but looks like the issue is with `resetFixtures`, which gets stuck at random tests.

I'll include the removed actions in a new PR and do some testing for that separately. This PR should be good to go now with the 2 new actions.

File: src/API.ts [hunk 1]
```diff
@@ -93,5 +92,7 @@ export async function apiRequest(
     instance.interceptors.response.use(_handleResponse, _handleError);
   }
 
-  return instance.request(axiosConfigAdapter(method, path, options));
+  return instance.request(
+    axiosConfigAdapter(method, stripExtraSlash(path), options),
```
Comment (sneridagh) [inline src/API.ts]: Are you sure we need this? In Volto we never had to do this. We can assume the paths will be correctly formed (eg. not ending in slash. In which moment the extra trailing slash is present?

File: src/restapi/actions/get.ts [hunk 2]
```diff
@@ -30,6 +30,6 @@ export const getActions = async ({
 };
 
 export const getActionsQuery = ({ path, config }: ActionsArgs) => ({
-  queryKey: [path, 'get', 'content'],
+  queryKey: [path, 'get', 'navigation'],
```
Comment (sneridagh) [inline src/restapi/actions/get.ts]: navigation? aren't this the actions?

**Review by sneridagh — CHANGES_REQUESTED** [review]

File: src/API.ts (hunk 1 above)
Comment (hemant-hc) [reply src/API.ts]: It was not necessary, I introduced it for the sake of maintaining clarity. I have removed it as per your suggestion.

File: src/restapi/actions/get.ts (hunk 2 above)
Comment (hemant-hc) [reply src/restapi/actions/get.ts]: It's fixed now.

**Comment by hemant-hc** [comment]
@sneridagh This PR should be good to go now.

File: src/restapi/actions/get.ts [hunk 3]
```diff
@@ -4,32 +4,32 @@ import { ActionsResponse } from '../../interfaces/actions';
...
+  actionId: z.string(),
 });
 
 export type ActionsArgs = z.infer<typeof getActionsSchema> & {
   config: PloneClientConfig;
 };
 
 export const getActions = async ({
-  path,
+  actionId,
```
Comment (sneridagh) [inline src/restapi/actions/get.ts]: I am confused with this one. You get the available actions per path (every path can have different actions)... you do not query actions IDs. Please refer to the restapi documentation.

File: src/restapi/actions/get.test.tsx [hunk 4]
```diff
@@ -24,10 +27,13 @@ describe('[GET] Actions', () => {
...
-    });
+    const actionId = 'blah';
+    const { result } = renderHook(
+      () => useQuery(getActionsQuery({ actionId })),
+      {
+        wrapper: createWrapper(),
+      },
+    );
 
     await waitFor(() => expect(result.current.isError).toBe(true));
```
Comment (sneridagh) [inline src/restapi/actions/get.test.tsx]: A bit more detailed test would have show you the nature of the endpoint.

File: src/restapi/addons/install.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    addonId,
+  });
+
+  const options: ApiRequestParams = {
+    config,
+    params: {},
+  };
+  const installAddonPath = `@addons/${validatedArgs.addonId}/install`;
+
+  return apiRequest('post', installAddonPath, options);
```
Comment (sneridagh) [inline src/restapi/addons/install.ts]: AddonName in all the cases... no need to differentiate them...

**Review by sneridagh — CHANGES_REQUESTED** [review]
Please take a look at my comments. For the rest is good to go once taken care.

**Review by sneridagh — APPROVED** [review]

**Pull request merged**